File: app/src/main/java/com/pin/recommend/model/viewmodel/AccountViewModel.java
package com.pin.recommend.model.viewmodel;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.pin.recommend.model.AppDatabase;
import com.pin.recommend.model.dao.AccountDao;
import com.pin.recommend.model.entity.Account;

public class AccountViewModel extends AndroidViewModel {

    private AccountDao dao;

    private LiveData<Account> accountLiveData;

    public AccountViewModel(Application application){
        super(application);
        dao = AppDatabase.getDatabase(application.getApplicationContext()).accountDao();
    }

    public LiveData<Account> getAccount(){
        initializeAccount();
        if(accountLiveData == null) {
           accountLiveData = dao.findTrackedById(Account.ACCOUNT_ID);
        }
        return accountLiveData;
    }

    private void initializeAccount(){
        AppDatabase.executor.execute(new Runnable() {
            @Override
            public void run() {
                Account model = dao.findById(Account.ACCOUNT_ID);
                if(model == null) {
                    model = new Account();
                    dao.insertAccount(model);
                }
            }
        });
    }

    public void saveAccount(final Account account){
        AppDatabase.executor.execute(new Runnable() {
            @Override
            public void run() {
                dao.updateAccount(account);
            }
        });
    }

}

File: app/src/main/java/com/pin/recommend/model/viewmodel/AccountViewModel.kt
package com.pin.recommend.model.viewmodel

import android.app.Application
import android.content.Context
import android.graphics.Color
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.Observer
import com.pin.recommend.model.AppDatabase
import com.pin.recommend.model.dao.AccountDao
import com.pin.recommend.model.entity.Account

class AccountViewModel(application: Application) : AndroidViewModel(application) {
    private val dao: AccountDao = AppDatabase.getDatabase(application.applicationContext).accountDao()
    val accountLiveData: LiveData<Account>

    init{
        initializeAccount()
        accountLiveData = dao.findTrackedById(Account.ACCOUNT_ID.toLong())
    }

    private fun initializeAccount() {
        AppDatabase.executor.execute {
            var model = dao.findById(Account.ACCOUNT_ID.toLong())
            if (model == null) {
                model = Account()
                dao.insertAccount(model)
            }
        }
    }

    fun saveAccount(account: Account?) {
        AppDatabase.executor.execute { dao.updateAccount(account) }
    }

}